packages/dds/tree/src/util/typeCheck.ts

```diff
@@ -193,6 +193,11 @@ export type requireAssignableTo<_A extends B, B> = true;
 
 /**
  * Returns a type parameter that is true iff the `Keys` union includes all the keys of `T`.
+ *
+ * @remarks
+ * This does not handle when the T has an index signature permitting keys like `string` which
+ * TypeScript cannot omit members from.
```

Comment:
Says does not handle but would be better to be more specific.
Maybe could say why we can't detect this unsupported case. (Seems easy to misuse in some complex case.)

Reply:
I avoided being specific on purpose. We could handle at least some such cases, but this version currently does not. I'd rather just keep them documented as unsupported so support could be added later if needed. I want to avoid investigating if supporting such cases is practical now (such an investigation is difficult and low value): we likely won't need it, and if we do later such an investigation can be done then.